Sources/PostgreSQL/Connection.swift

```diff
@@ -171,6 +180,37 @@ public final class Connection: ConnInfoInitializable {
         }
     }
     
+    /// Creates a dispatch read source for this connection that will call `callback` on `queue` when a notification is received
+    ///
+    /// - Parameter channel: the channel to register for
+    /// - Parameter queue: the queue to create the DispatchSource on
+    /// - Parameter callback: the callback
+    /// - Parameter note: The notification received from the database
+    /// - Parameter err: Any error while reading the notification. If not nil, the source will have been canceled
+    /// - Returns: the dispatch socket to activate
+    /// - Throws: if fails to get the socket for the connection
+    public func makeListenDispatchSource(toChannel channel: String, queue: DispatchQueue, callback: @escaping (_ note: Notification?, _ err: Error?) -> Void) throws -> DispatchSourceRead {
+        guard let sock = Optional.some(PQsocket(self.cConnection)), sock >= 0
+            else { throw PostgreSQLError(code: .ioError, reason: "failed to get socket for connection") }
+        let src = DispatchSource.makeReadSource(fileDescriptor: sock, queue: queue)
+        src.setEventHandler { [unowned self] in
+            do {
+                try self.validateConnection()
+                PQconsumeInput(self.cConnection)
+                while let pgNotify = PQnotifies(self.cConnection) {
+                    let notification = Notification(pgNotify: pgNotify.pointee)
+                    callback(notification, nil)
+                    PQfreemem(pgNotify)
+                }
+            } catch {
+                callback(nil, error)
+                src.cancel()
+            }
+        }
+        try self.execute("LISTEN \(channel)")
+        return src
+    }
+    
     /// Registers as a listener on a specific notification channel.
```

Comment:
Considering the new method should be the way to go, I would mark the old function as deprecated.